src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -7130,24 +7140,187 @@ public override IOperation VisitArgument(IArgumentOperation operation, int? capt
+
+                    var property = propertyReference.Property;
+                    if (explicitProperties.ContainsKey(property))
+                    {
+                        AddStatement(VisitRequired(simpleAssignment.Value));
+                        continue;
+                    }
+
+                    int valueCaptureId = GetNextCaptureId(outerCaptureRegion);
+                    VisitAndCapture(simpleAssignment.Value, valueCaptureId);
+                    LeaveRegionsUpTo(innerCaptureRegion);
+                    var valueCaptureRef = new FlowCaptureReferenceOperation(valueCaptureId, operation.Operand.Syntax,
+                        operation.Operand.Type, constantValue: operation.Operand.GetConstantValue());
+
+                    var assignment = makeAssignment(property, valueCaptureRef, operation);
+
+                    explicitProperties.Add(property, assignment);
+                }
+
+                // Make a sequence for all properties (in order), constructing assignments for the implicitly set properties
+                var type = (INamedTypeSymbol)operation.Type;
+                foreach (IPropertySymbol property in properties)
+                {
+                    if (explicitProperties.TryGetValue(property, out var assignment))
+                    {
+                        initializerBuilder.Add(assignment);
+                    }
+                    else
+                    {
+                        Debug.Assert(oldValueCaptureId >= 0);
+
+                        // `oldInstance`
+                        var oldInstance = new FlowCaptureReferenceOperation(oldValueCaptureId, operation.Operand.Syntax,
+                            operation.Operand.Type, constantValue: operation.Operand.GetConstantValue());
+
+                        // `oldInstance.Property`
+                        var visitedValue = new PropertyReferenceOperation(property, ImmutableArray<IArgumentOperation>.Empty, oldInstance,
+                            semanticModel: null, operation.Syntax, property.Type, isImplicit: true);
+
+                        int extraValueCaptureId = GetNextCaptureId(outerCaptureRegion);
+                        AddStatement(new FlowCaptureOperation(extraValueCaptureId, operation.Syntax, visitedValue));
+
+                        var extraValueCaptureRef = new FlowCaptureReferenceOperation(extraValueCaptureId, operation.Operand.Syntax,
+                            operation.Operand.Type, constantValue: operation.Operand.GetConstantValue());
+
+                        assignment = makeAssignment(property, extraValueCaptureRef, operation);
+                        initializerBuilder.Add(assignment);
+                    }
+                }
+                LeaveRegionsUpTo(outerCaptureRegion);
+
+                return new AnonymousObjectCreationOperation(initializerBuilder.ToImmutableAndFree(), semanticModel: null, operation.Syntax, operation.Type, operation.IsImplicit);
```

Comment:
>initializerBuilder [](http://example.com/codeflow?start=60&length=18)

Consider adding an assert that the amount of initializers matches the amount of properties.